freeipa/src/main/java/com/sequenceiq/freeipa/service/image/ImageCatalogProvider.java

```diff
@@ -47,48 +47,58 @@
     @Value("#{'${cb.enabled.linux.types}'.split(',')}")
     private List<String> enabledLinuxTypes;
 
+    @Value("${info.app.version:}")
+    private String freeIpaVersion;
+
     @Inject
     private ObjectMapper objectMapper;
 
     @Cacheable(cacheNames = "imageCatalogCache", key = "#catalogUrl")
     public ImageCatalog getImageCatalog(String catalogUrl)  {
-        ImageCatalog catalog;
-        if (catalogUrl == null) {
-            LOGGER.info("No image catalog was defined!");
-            return null;
-        }
-
         try {
-            long started = System.currentTimeMillis();
-            String content;
-            if (catalogUrl.startsWith("http")) {
-                Client client = RestClientUtil.get();
-                WebTarget target = client.target(catalogUrl);
-                Response response = target.request().get();
-                content = readResponse(target, response);
-            } else {
-                content = readCatalogFromFile(catalogUrl);
+            if (Objects.nonNull(catalogUrl)) {
+                long started = System.currentTimeMillis();
+                String content = readCatalogContent(catalogUrl);
+                ImageCatalog catalog = objectMapper.readValue(content, ImageCatalog.class);
+                if (Objects.nonNull(catalog)) {
+                    ImageCatalog filteredCatalog = filterImagesByOsType(catalog);
+                    long timeOfParse = System.currentTimeMillis() - started;
+                    LOGGER.debug("ImageCatalog was fetched and parsed from '{}' and took '{}' ms.", catalogUrl, timeOfParse);
+                    return filteredCatalog;
+                }
+                throw new ImageCatalogException(String.format("Failed to read an image catalog from '%s' content.", content));
             }
-            catalog = filterImagesByOsType(objectMapper.readValue(content, ImageCatalog.class));
-            long timeOfParse = System.currentTimeMillis() - started;
-            LOGGER.debug("ImageCatalog was fetched and parsed from '{}' and took '{}' ms.", catalogUrl, timeOfParse);
+            throw new ImageCatalogException("Unable to fetch image catalog by the catalogUrl is null.");
```

Comment:
This message looks strange. Can you please rewrite it to `Unable to fetch image catalog. The catalogUrl is null.`?